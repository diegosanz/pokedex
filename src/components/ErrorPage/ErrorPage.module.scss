.error-page {
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;

  & .home-link {
    color: blue;
    text-decoration: underline;
  }
}
