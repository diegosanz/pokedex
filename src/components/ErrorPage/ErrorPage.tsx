import { FC } from 'react'
import { Link } from 'react-router-dom'
import notFoundImg from '../../assets/img/404.jpg'
import styles from './ErrorPage.module.scss'

const ErrorPage: FC = () => {
  return (
    <div className={styles.errorPage}>
      <img src={notFoundImg} alt="404" />

      <Link to={'/'} className={styles.homeLink}>
        Go to the main page
      </Link>
    </div>
  )
}

export default ErrorPage
