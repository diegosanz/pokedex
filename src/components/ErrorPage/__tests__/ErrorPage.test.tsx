import { render, screen } from '@testing-library/react'
import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import ErrorPage from '../ErrorPage'

describe('ErrorPage', async () => {
  const setup = () => {
    return render(
      <BrowserRouter>
        <ErrorPage />
      </BrowserRouter>
    )
  }

  it('Renders correctly and unmount correctly', () => {
    const { unmount } = setup()
    unmount()
  })

  it('Match Snapshot', () => {
    const { container } = setup()
    expect(container).toMatchSnapshot()
  })

  it('Should contain error title', () => {
    setup()

    expect(screen.getByRole('link')).toHaveTextContent(/Go to the main page/i)
  })
})
