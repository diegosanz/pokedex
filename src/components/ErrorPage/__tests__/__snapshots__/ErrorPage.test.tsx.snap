// Vitest Snapshot v1

exports[`ErrorPage > Match Snapshot 1`] = `
<div>
  <div
    class="_errorPage_8c5621"
  >
    <img
      alt="404"
      src="/src/assets/img/404.jpg"
    />
    <a
      class="_homeLink_8c5621"
      href="/"
    >
      Go to the main page
    </a>
  </div>
</div>
`;
